### PR TITLE
[BACKLOG-6049] boolean, function, object, and string specs

### PR DESCRIPTION
**Pull request by scottyaslan**
@dcleao @carlosrusso

### CODE REVIEW

File: package-res/resources/web/test/karma/unit/pentaho/type/boolean.Spec.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                PentahoBoolean = booleanFactory(new Context());
+            });
+            it("should return an object", function () {
+                expect(typeof new PentahoBoolean(1)).toBe("object");
+            });
+            it("should return a function", function () {
+                expect(typeof PentahoBoolean).toBe("function");
+            });
+            it("should accept 1 as true", function () {
+                expect(new PentahoBoolean(1)._value).toBe(true);
```
Comment (carlosrusso) [inline package-res/resources/web/test/karma/unit/pentaho/type/boolean.Spec.js]: please use the getter interfaces directly and the valueOf method:

expect(new PentahoBoolean(1).value).toBe(true); //getter
expect(new PentahoBoolean(1).v).toBe(true); // shorter version of .value getter
expect(new PentahoBoolean(1).valueOf()).toBe(true);

File: package-res/resources/web/test/karma/unit/pentaho/type/boolean.Spec.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                expect(function () {
+                    new PentahoBoolean(null)
+                }).toThrowError(bundle.structured.errors.value.isNull);
+            });
+            it("should not accept undefined", function () {
+                expect(function () {
+                    new PentahoBoolean(undefined)
+                }).toThrowError(bundle.structured.errors.value.isNull);
+            });
+        });
```
Comment (carlosrusso) [inline package-res/resources/web/test/karma/unit/pentaho/type/boolean.Spec.js]: Personally, I would also add some "sad path" cases:

expect(new PentahoBoolean("potatoes").value).toBe(false); //or .toBeNull(), whatever applies
expect(new PentahoBoolean(new Date()).value).toBe(false);

File: package-res/resources/web/test/karma/unit/pentaho/type/string.Spec.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                expect(function () {
+                    new PentahoString(null)
+                }).toThrowError(bundle.structured.errors.value.isNull);
+            });
+            it("should not accept undefined", function () {
+                expect(function () {
+                    new PentahoString(undefined)
+                }).toThrowError(bundle.structured.errors.value.isNull);
+            });
+        });
```
Comment (carlosrusso) [inline package-res/resources/web/test/karma/unit/pentaho/type/string.Spec.js]: expect(new PentahoString({}).value).toBe ?
expect(new PentahoString([]).value).toBe ?

File: package-res/resources/web/test/karma/unit/pentaho/type/boolean.Spec.js (hunk 1 above)
Comment (dcleao) [inline package-res/resources/web/test/karma/unit/pentaho/type/boolean.Spec.js]: Hi, the `v` property is also an internal detail used for de-serialization/configuration purposes only. Please use the `value` property in this context.
`valueOf()` will be tested in the `Simple` tests.

File: package-res/resources/web/test/karma/unit/pentaho/type/string.Spec.js (hunk 3 above)
Comment (dcleao) [inline package-res/resources/web/test/karma/unit/pentaho/type/string.Spec.js]: `PentahoString({}).value` should throw cause it is taken as a spec and the value is not there, thus it is `null` :-)

**Pull request merged**